contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2093,6 +2121,12 @@ static void process_query_specification(
 			}
 		}
 	}
+	
+	if(query_hints.size()) {
+		add_query_hints(expr);
```

Comment:
Not sure process_query_specification is the right place to add query hints. Please check the following cases:
- INSERT/DELETE/UPDATE statements
- Multiple select statements in a single query, e.g., UNION and CTE (Common Table Expressions)

Reply:
Moved the change to insert the hints at a common place for all queries